vignettes/minimal-rstan-package.Rmd

```diff
@@ -0,0 +1,160 @@
+---
+title: "A working example for writing a package that depends on RStan"
+author: "Stefan Siegert"
+date: "`r Sys.Date()`"
+output:
+  rmarkdown::html_vignette:
+    toc: true
+vignette: >
+  %\VignetteIndexEntry{Minimal RStan Package}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, echo=FALSE, eval=TRUE}
+knitr::opts_chunk$set(eval=FALSE)
+```
+
+## Creating a minimal package that uses Stan, step-by-step
+
+This vignette is a step-by-step guide to create a functioning R package that uses Stan.
+It is complementary to the [rstantools developer guidelines vignette](developer-guidelines.html) which you should read first.
+
+
+To start off, we use `rstan_package_skeleton` to initialise a bare-bones package directory.
+The name of our package will be `rstanlm`; it will fit a simple linear regression model using Stan.
+Under the hood, the function `rstan_package_skeleton` calls `utils::package.skeleton()` which won't allow creating a completely empty package directory, so we initialise it with a dummy variable which we will delete later on:
+
+```{r}
+library(rstantools)
+rstan_package_skeleton(name='rstanlm', environment=as.environment(list(dummy=42)))
+```
+
+The newly created package directory is called `rstanlm` and we change to it to make a few modifications:
+
+```{r}
+setwd('rstanlm')
+```
+
+We remove the following files, which are either unneeded, or will be automatically recreated later on:
+
+```{r}
+file.remove('Read-and-delete-me')
+file.remove('data/dummy.rda')
+file.remove('man/dummy.Rd')
```

Comment:
should't be necessary if not including the dummy environment in the call to `rstan_package_skeleton`